tutorial/stream-client.md

```diff
@@ -0,0 +1,276 @@
+
+(sec-create-access-token)=
+### Create an access token
+
+To access the HTTP API of the Anbox Cloud stream gateway, an access token is required. Each access token is associated with a service account.
+
+On the machine where Anbox Cloud Appliance is installed, create the service account by running the following command:
+
+    sudo anbox-cloud-appliance gateway account create streaming-tutorial
+
+The output of this command provides the access token. Make a note of this token to use when you make a request to the stream gateway API.
+
+See {ref}`howto-access-stream-gateway` for more information on creating, using and deleting the access token.
+
+### Create an Android Instance
+
+Create a streaming enabled Android instance:
+
+    arch="$(dpkg-architecture -q DEB_HOST_ARCH)"
+    amc launch --enable-streaming --raw --name a13 jammy:android13:"$arch"
+
+### Determine session ID of the Android instance
+
+For the next step, we need the session ID which AMS assigned to the instance. So let's get that using the following command:
+
+    amc ls --filter name=a13 --format=csv | cut -d, -f6 | awk -F"session=" '{ print $2 }'
+
+## Implement the stream client
+
+Now that we have everything ready, let's create a directory to set up the stream client:
+
+    mkdir -p stream-client/{templates,static}
+
+We will be using [Flask](https://flask.palletsprojects.com/en/stable/) for the purpose of this tutorial. First, let us create a `requirements.txt` file inside `stream-client` with the following code:
+
+```
+flask==3.0.3
+requests==2.32.3
+pyopenssl==24.2.1
+```
+
+Create a `service.py` file inside `stream-client` with the following code:
+
+```python
+import os
+import os.path
+import secrets
+
+from flask import Flask, jsonify, render_template, request
+from requests import Session
+from functools import wraps
+from typing import *
+
+
+class GatewayAPI:
+    def __init__(self, base_url, session, token, verify):
+        self.base_url = base_url
+        self.session = session
+        self.token = token
+        self.verify = verify
+
+    def make_request(
+        self,
+        method,
+        path,
+        headers,
+        data,
+    ):
+        uri = f"{self.base_url}{path}"
+
+        headers["Authorization"] = f"macaroon root={self.token}"
+        headers["Content-Type"] = "application/json"
+
+        return self.session.request(
+            method, uri, json=data, headers=headers, verify=self.verify
+        )
+
+    def join_session(self, session_id, data):
+        uri = f"/1.0/sessions/{session_id}/join"
+        return self.make_request(
+            "POST",
+            uri,
+            data=data,
+            headers={"Content-Type": "application/json"},
```

Comment:
Is this needed? otherwise you will override the `Content-Type` header in the make_request function.

Reply:
@adglkh You mean to say that the header declaration line is unnecessary or that it will cause errors?